src/redis/driver_impl.go

```diff
@@ -45,78 +49,53 @@ func checkError(err error) {
 	}
 }
 
-func (this *poolImpl) Get() Connection {
-	client, err := this.pool.Get()
-	checkError(err)
-	this.stats.connectionActive.Inc()
-	this.stats.connectionTotal.Inc()
-	return &connectionImpl{client, 0}
-}
+func NewClientImpl(scope stats.Scope, useTls bool, auth string, url string, poolSize int,
+	pipelineWindow time.Duration, pipelineLimit int) Client {
+	logger.Warnf("connecting to redis on %s with pool size %d", url, poolSize)
 
-func (this *poolImpl) Put(c Connection) {
-	impl := c.(*connectionImpl)
-	this.stats.connectionActive.Dec()
-	if impl.pending == 0 {
-		this.pool.Put(impl.client)
-	} else {
-		// radix does not appear to track if we attempt to put a connection back with pipelined
-		// responses that have not been flushed. If we are in this state, just kill the connection
-		// and don't put it back in the pool.
-		impl.client.Close()
-		this.stats.connectionClose.Inc()
-	}
-}
+	df := func(network, addr string) (radix.Conn, error) {
+		var dialOpts []radix.DialOpt
 
-func NewPoolImpl(scope stats.Scope, useTls bool, auth string, url string, poolSize int) Pool {
-	logger.Warnf("connecting to redis on %s with pool size %d", url, poolSize)
-	df := func(network, addr string) (*redis.Client, error) {
-		var conn net.Conn
 		var err error
 		if useTls {
-			conn, err = tls.Dial("tcp", addr, &tls.Config{})
-		} else {
-			conn, err = net.Dial("tcp", addr)
+			dialOpts = append(dialOpts, radix.DialUseTLS(&tls.Config{}))
```

Comment:
might be a dumb question. For non-tls, non-auth case, the dialOpts would be empty, is it still able to connect to redis?

Reply:
Yes.